inbox/mailsync/service.py

```diff
@@ -118,24 +117,24 @@ def run(self):
         while self.keep_running:
             retry_with_logging(self._run_impl, self.log)
 
+        kill_all(self.contact_sync_monitors.values())
+        self.log.info(
+            "stopped contact sync monitors", count=len(self.contact_sync_monitors)
+        )
+        kill_all(self.event_sync_monitors.values())
+        self.log.info(
+            "stopped event sync monitors", count=len(self.event_sync_monitors)
+        )
+
```

Comment:
`event_sync_monitors` and `contact_sync_monitors` are easier/faster to kill because they are just single greenlets as opposed to trees of greenlets in `email_sync_monitors`. It makes sense to kill them first because of that. The order is not really meaningful from correctness point of view.